### ISSUE TITLE
Changes to published/draft status badge in search results

### ISSUE THREAD

**Issue by srerickson** [psu-libraries/scholarsphere#1010]
# Feature Request

In search results and on the dashboard, each entry has a "badge" indicating the work versions's published/draft status: 
![badge](https://user-images.githubusercontent.com/876323/117307731-d211dc00-ae4e-11eb-9dd2-6a1910186266.jpg)

Currently, this badge reflects the state of the most recent work version in a work. This feature requests changes the badge behavior to reflect the state of the parent work instead of the individual work version:

- An entry should appear as "published" if the parent work has at least one published version.
- An entry should appear as "withdrawn" if the parent work has no published versions and at least one withdrawn version.
- An entry should appear as "draft" if the parent work only has versions in draft state. (Because a work may only have one draft version at a time, there should be only one version in this case).

Relates to discussion in #441 
Blocked by #1026

**Comment by awead** [comment]
@nicktiberi I've moved this to blocked status because we need to change the indexing. I think it will solve a lot of other problems

**Comment by awead** [comment]
This is working now due to #1037, but it still needs to tidying up. See below

![Screen Shot 2021-05-19 at 11 34 25 AM](https://user-images.githubusercontent.com/312085/118841457-5411f400-b896-11eb-8c72-a38f3d8450a2.png)

The withdrawn badge is right up against the border. Maybe shifting all the badges over to the left a couple of pixels?

**Comment by nicktiberi** [comment]
@awead The problem here is that the badges are overlapping the available column width. I could fix that in a separate PR if you'd like.

<img width="239" alt="Screen Shot 2021-05-19 at 11 55 20 AM" src="https://user-images.githubusercontent.com/639920/118844646-1cf11200-b899-11eb-9ddf-6a81447d7baa.png">

**Comment by nicktiberi** [comment]
Opened #1040 to address the spacing/overlap issues.

**Comment by awead** [comment]
@nicktiberi I think we can keep this ticket open and you can fix it here, if you like.
